app/src/main/java/io/gnosis/kouban/di/FragmentModule.kt

```diff
@@ -71,4 +74,11 @@ val fragmentModule = module {
         }
         factory { BaseAdapter(TransactionDetailFactory()) }
     }
+
+    scope(named<BalancesWidgetConfigure>()) {
+        viewModel { BalancesViewModel(get(), get(), get()) }
+        factory { BalancesItemFactory(get(), get()) }
+        factory { (onTokenClickedListener: BalancesItemFactory.OnTokenClickedListener) ->
+            BaseAdapter(BalancesItemFactory(get(), onTokenClickedListener)) }
```

Comment:
Oh man I would feel very uncomfortable about injecting a listener. Specially if the listener is an activity like in this case. I would suggest either setting the listeners directly in your viewHolder or maybe using the viewModel instead for this.

Reply:
I think injecting a listener is fine, but you would need to make sure that it doesn't retain the activity (maybe use a weak reference)